src/core/utils/helpers/modal-helpers.ts

```diff
@@ -45,3 +45,9 @@ export const getDetailsModalId = (queryParam: string, prefix: string) => {
   }
   return "";
 };
+
+export const createLoanModalId = (
+  dueDate: string | null | undefined,
+  dueDateModal: string | number | Record<string, unknown>,
+  allLoansId: string | number | Record<string, unknown>
+) => (dueDate ? `${dueDateModal}-${dueDate}` : String(allLoansId));
```

Comment:
I think we should define modalId functions in the modal components.
That makes it more modular. If we want to delete the modal in the future the whole thing is self-conatined.
Plus that if there is some functionality involved in finding out the omdal id fragments, it probably comes from functionality inside of the component itself.

Reply:
For consistency I think you should use this function when creating the id:
https://github.com/danskernesdigitalebibliotek/dpl-react/blob/4f75d55e348f0395031fa4d8d76397ece159e504/src/core/utils/helpers/modal-helpers.ts#L3-L5

Reply:
@spaceo but we've been creating this specific modal id differently. I just took the function out into a helper.

Reply:
I don't think you need a full overhaul. Just like it has been done eg. here:
https://github.com/danskernesdigitalebibliotek/dpl-react/blob/4f75d55e348f0395031fa4d8d76397ece159e504/src/components/find-on-shelf/FindOnShelfModal.tsx#L12-L14

Reply:
donezies :)

In regards to the self-containment of modal related stuff - I think it's pretty self contained when the function exists in a file called `modal-helpers.ts`